vault/identity_store_entities.go

```diff
@@ -799,23 +861,41 @@ func (i *IdentityStore) mergeEntity(ctx context.Context, txn *memdb.Txn, toEntit
 			return errors.New("entity id to merge from does not belong to this namespace"), nil
 		}
 
-		for _, alias := range fromEntity.Aliases {
+		for _, fromAlias := range fromEntity.Aliases {
+			if toAliasId, ok := toEntityAccessors[fromAlias.MountAccessor]; ok {
+				// Handle conflicts (conflict = both aliases share the same mount accessor)
+				if strutil.StrListContains(conflictingAliasIDsToKeep, toAliasId) {
+					i.logger.Info("Deleting from_entity alias during entity merge", "from_entity", fromEntityID, "deleted_alias", fromAlias.ID)
+					err := i.MemDBDeleteAliasByIDInTxn(txn, fromAlias.ID, false)
+					if err != nil {
+						return nil, fmt.Errorf("failed to delete orphaned alias during merge: %w", err)
+					}
+
+					// Continue to next alias, as there's no alias to merge left in the from_entity
+					continue
```

Comment:
Why is this continue necessary?

Reply:
We have two paths:
- Keep the alias in the toEntity
- Keep the alias in the fromEntity

The continue is part of the toEntity path. In short, since we delete alias in the fromEntity, there's nothing to do to merge, as the alias we want to be in the toEntity is already there.

In the second path, there's a bit of code to actually assign the fromEntity's alias to the toEntity, but we don't need that in the first path's case.